lib/improver/utilities/statistical_operations.py

```diff
@@ -0,0 +1,256 @@
+        if (first_percentile - last_percentile >= 0).all():
+            self.inverse_ordering = True
+
+    def __repr__(self):
+        """Represent the configured plugin instance as a string."""
+        result = ('<ProbabilitiesFromPercentiles2D: percentiles_cube: {}, '
+                  'output_name: {}'.format(self.percentiles_cube,
+                                           self.output_name))
+        return result
+
+    def create_probability_cube(self, cube):
+        """
+        Create a 2-dimensional probability cube in which to store the
+        calculated probabilities.
+
+        Args:
+            cube (iris.cube.Cube):
+                Template for the output probability cube.
+        Returns:
+            probability_cube (iris.cube.Cube):
+                A new 2-dimensional probability cube with suitable metadata.
+        """
+        cube_format = next(cube.slices([cube.coord(axis='y'),
+                                        cube.coord(axis='x')]))
+        probabilities = cube_format.copy(data=np.full(cube_format.shape,
+                                                      np.nan, dtype=float))
+        try:
+            probabilities.remove_coord(self.percentile_coordinate)
+        except CoordinateNotFoundError:
+            pass
+
+        probabilities.units = 1
+        probabilities.rename(self.output_name)
+        return probabilities
+
+    def percentile_interpolation(self, threshold_cube, percentiles_cube):
+        """
+        Using a percentiles_cube containing a distinct percentile distribution
+        for each point on a 2-dimensional grid, we can interpolate through each
+        distribution to obtain a probability. The point to which we interpolate
+        is defined by the threshold_cube.
+
+        Args:
+            threshold_cube (iris.cube.Cube):
+                A 2-dimensional cube of "threshold" values for which it is
+                desired to obtain probability values from the percentiled
+                reference cube.
+            percentiles_cube (iris.cube.Cube):
+                A 3-dimensional cube, 1 dimension describing the percentile
+                distributions, and 2-dimensions shared with the threshold_cube,
+                typically x and y.
+        Returns:
+            probabilities (iris.cube.Cube):
+                A 2-dimensional cube of probabilities obtained by interpolating
+                between percentile values.
+        """
+        percentiles = self.percentile_coordinate.points
+        probabilities = self.create_probability_cube(percentiles_cube)
+
+        array_shape = [2] + list(threshold_cube.shape)
+        percentile_bounds = np.full(array_shape, -1, dtype=float)
+        value_bounds = np.full(array_shape, np.nan, dtype=float)
+
+        for index, pslice in enumerate(percentiles_cube.slices_over(
+                self.percentile_coordinate)):
+            # Change to use < and > to force degernate percentile distributions
```

Comment:
We probably need more description in one of the main docstrings about the behaviour for degenerate values in your distribution.